### PR TITLE
add argo cd guide

### LINKED ISSUE #2121: XRC -> XR Label Propagation Confuses ArgoCD
Edit from @negz:

> In the case of an XR created by an XRC, ArgoCD sees a XR appear with the application ownership label (currently automatically propagated by Crossplane), but without an owner reference to any known resource

I believe, per @mcavoyk's comment above, that this is the key issue we're seeing here. ArgoCD thinks _it_ (not Crossplane) created the XR, because when Crossplane creates the XR it propagates a label ArgoCD uses to track which resource it owns. This results in the XR either being shown as "out-of-sync" or being deleted depending on whether auto-prune is enabled.

Original issue follows.

---

Dear all,
in the scenario we are testing we deploy on a k8s cluster crossplane and all the resources taked from [platform-ref-gcp](https://github.com/upbound/platform-ref-gcp) except the final resource of the example folder.
These last resources  for example triggering the [Cluster creation](https://github.com/upbound/platform-ref-gcp/blob/main/examples/cluster.yaml) was done using ArgoCD that formally connect to the k8s cluster and execute the yaml.
Flagging the [auto-pruning feature]( https://argoproj.github.io/argo-cd/user-guide/auto_sync/) of ArgoCD leads to a sort of resource contentions between Crossplane and Argo:

`Normal   ConfigureCompositeResource  19s (x3 over 20s)  offered/compositeresourcedefinition.apiextensions.crossplane.io  Successfully applied composite resource
  Warning  ConfigureCompositeResource  19s (x2 over 20s)  offered/compositeresourcedefinition.apiextensions.crossplane.io  cannot patch object: Operation cannot be fulfilled on compositenetworks.gcp.platformref.crossplane.io "test-network-d6xjh": the object has been modified; please apply your changes to the latest version and try again
  Normal   BindCompositeResource       18s (x3 over 20s)  offered/compositeresourcedefinition.apiextensions.crossplane.io  Composite resource is not yet ready
  Warning  BindCompositeResource       18s                offered/compositeresourcedefinition.apiextensions.crossplane.io  compositenetworks.gcp.platformref.crossplane.io "test-network-d6xjh" not found`

and the Cluster was not created.
Disabling auto-pruning indeed lead to the resource creation, but Argo remain out-of-sync.
In my opinion the correct ways is to having all the crd and definiton of gcp-reference done by Argo, not only the last one.
So I am not sure if is actually a bug or not, or simple an usage misunderstanding.
Thanks,
Ivan

### PR DESCRIPTION
**Pull request by natereid72**
### Description of your changes

Adding guide to cover configuring Argo CD with Crossplane

This documentation should fix #2121, in the sense that the community will have the resource necessary to configure Argo to play well with Crossplane

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

No testing. Doc update.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by natereid72** [comment]
@hasheddan This was a request from @luebken. I couldn't label the backport. but hope the rest is good.

**Comment by natereid72** [comment]
@jbw976

File: docs/guides/argo-cd-crossplane.md
Comment (negz) [inline docs/guides/argo-cd-crossplane.md:71]: We should probably use upstream Crossplane rather than UXP on https://crossplane.io docs.

Comment (natereid72) [reply docs/guides/argo-cd-crossplane.md:71]: @negz Yep, that was on my to-do. I got confused and thought the reviewer made that change. If it's merged already, I'll rePR it with fix

Comment (natereid72) [reply docs/guides/argo-cd-crossplane.md]: Ok, didn't realize I had the additional ref to an Upbound chart. I just removed that portion of the issue entirely because it seems to be a preference type of config. Left with most common issue encountered and all directly Crossplane focused.

Comment (natereid72) [reply docs/guides/argo-cd-crossplane.md]: @negz Looks like I didn't realize this one was still open and doubled up on another doc change. Let me know if it's ok to merge as-is or if I need to break these apart with feature branches.

File: docs/guides/argo-cd-crossplane.md
Comment (negz) [inline docs/guides/argo-cd-crossplane.md:37]: Somewhere in this guide we should link to https://github.com/crossplane/crossplane/issues/2773 and clarify that this is not "working as intended" - i.e. that we desire a better experience than this.

File: docs/guides/vault-as-secret-store.md
Comment (negz) [inline docs/guides/vault-as-secret-store.md:152]: We'll need to remove this mention of UXP as well.

File: docs/guides/argo-cd-crossplane.md
Comment (negz) [reply docs/guides/argo-cd-crossplane.md]: @natereid72 I think you just need to address the two other comment threads below and this should be good to merge. I'll close this one out.

File: docs/guides/vault-as-secret-store.md
Comment (negz) [reply docs/guides/vault-as-secret-store.md:152]: @natereid72 This and one other reference to UXP need to be removed before this can be merged.

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:20]:
```suggestion
with those running in a Kubernetes cluster (GitOps). There are different ways to configure 

```

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:16]:
```suggestion
This doc will help you understand these requirements. It is recommended to use

```

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:17]: Is the version limitation driven by a specific feature in this release?

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:27]:
```suggestion
`ConfigMap` in the `argocd` `Namespace`. Add `application.resourceTrackingMethod: annotation`

```

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:26]:
```suggestion
To configure Argo CD for Annotation resource tracking, edit the `argocd-cm`

```

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:30]:
```suggestion
```yaml

```

File: docs/guides/argo-cd-crossplane.md
Comment (ytsarev) [inline docs/guides/argo-cd-crossplane.md:37]:
```suggestion
On the next Argo CD sync, Crossplane `Claims` and `Composite Resources` will

```

File: docs/guides/vault-as-secret-store.md
Comment (ytsarev) [inline docs/guides/vault-as-secret-store.md:152]:
```suggestion
cat << EOF > values.yaml
```

File: docs/guides/vault-as-secret-store.md
Comment (ytsarev) [inline docs/guides/vault-as-secret-store.md:161]:
```suggestion
helm upgrade --install crossplane crossplane-stable/crossplane --namespace crossplane-system -f values.yaml
```

**Review by ytsarev — CHANGES_REQUESTED** [review]
@natereid72 please consider review suggestions

File: docs/guides/argo-cd-crossplane.md
Comment (natereid72) [reply docs/guides/argo-cd-crossplane.md:18]: it's the version I tested with. It probably works as of 2.2.0, but I didn't load all of those and test.

Comment (ytsarev) [reply docs/guides/argo-cd-crossplane.md:18]: cool, makes sense

**Review by ytsarev — APPROVED** [review]
thanks for the fixes, lgtm

**Comment by ytsarev** [comment]
@natereid72 argh, looks like DCO is broken after GitHub suggestion-based commit. You'll need to add 'Signed-off-by' to them

**Comment by negz** [comment]
> DCO is broken after GitHub suggestion-based commit. You'll need to add 'Signed-off-by' to them

I recommend squashing these all into one commit, and signing that one.

File: docs/guides/argo-cd-crossplane.md
Comment (almereyda) [inline docs/guides/argo-cd-crossplane.md:17]: Is there a special reason why this blank line breaks the sentence here?

**Comment by negz** [comment]
@natereid72 Since you had signed some of your commits I used GitHub's "Squash and Merge" functionality to squash them into one big signed commit.

**Pull request merged**
